apps/aechannel/src/aesc_offchain_state.erl

```diff
@@ -398,3 +409,44 @@ serialize_for_client_tx_or_notx(Tx) ->
     STx = aetx_sign:serialize_to_binary(Tx),
     aeser_api_encoder:encode(transaction, STx).
 
+-spec serialize_to_binary(state()) -> binary().
+serialize_to_binary(#state{ trees = Trees
+                          , calls = Calls
+                          , signed_tx = SignedTx
+                          , half_signed_tx = HalfSignedTx
+                          }) ->
+        aeser_rlp:encode([
+            aec_trees:serialize_to_binary(Trees),
+            aect_call_state_tree:to_binary_without_backend(Calls),
+            serialize_to_binary_tx_or_notx(SignedTx),
+            serialize_to_binary_tx_or_notx(HalfSignedTx)
+        ]).
+
+-spec serialize_to_binary_tx_or_notx(aetx_sign:signed_tx() | ?NO_TX) -> binary().
+serialize_to_binary_tx_or_notx(?NO_TX) ->
+    <<"">>;
+serialize_to_binary_tx_or_notx(Tx) ->
+    aetx_sign:serialize_to_binary(Tx).
+
+-spec deserialize_from_binary(binary()) -> state().
+deserialize_from_binary(Bin) ->
+    case aeser_rlp:decode(Bin) of
+        [ BinTrees
+        , BinCalls
+        , BinSignedTx
+        , BinHalfSignedTx
+        ] ->
+            #state{ trees = aec_trees:deserialize_from_binary_without_backend(BinTrees)
+                  , calls = aect_call_state_tree:from_binary_without_backend(BinCalls)
+                  , signed_tx = deserialize_tx_or_notx_from_binary(BinSignedTx)
+                  , half_signed_tx = deserialize_tx_or_notx_from_binary(BinHalfSignedTx)
+                  };
+        _ ->
+            erlang:error("Failed to deserialize offchain state")
```

Comment:
I believe the convention is to pass an atom to the `error/1` function. If one wants this pretty-printed, this can be done in a separate function.

Reply:
I agree.

Reply:
I will make a PR with a one line fix later - right now trying to get this pr merged before the merge window for 5.0.0-rc2 closes.